src/hooks/useBrowseResults.ts

```diff
@@ -1,18 +1,65 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useReducer } from 'react';
+import {
+  Nullable,
+  IBrowseParameters as BrowseParameters,
+  GetBrowseResultsResponse,
+} from '@constructor-io/constructorio-client-javascript/lib/types';
 import { useCioPlpContext } from './useCioPlpContext';
 import useRequestConfigs from './useRequestConfigs';
 import { transformBrowseResponse } from '../utils/transformers';
-import { PlpBrowseData } from '../types';
+import { ConstructorIOClient, PlpBrowseData } from '../types';
 import { getBrowseParamsFromRequestConfigs } from '../utils';
 import useFirstRender from './useFirstRender';
+import {
+  RequestStatus,
+  requestReducer,
+  RequestAction,
+  initialState,
+  initFunction,
+  RequestType,
+} from '../components/CioPlpGrid/reducer';
 
 export interface UseBrowseResultsProps {
-  initialBrowseResponse?: PlpBrowseData;
+  initialBrowseResponse?: GetBrowseResultsResponse;
 }
 
 export type UseBrowseResultsReturn = {
-  browseResults: PlpBrowseData | null;
-  handleSubmit: () => void;
+  status: RequestStatus | null;
+  message?: string;
+  data: Nullable<PlpBrowseData>;
+  refetch: () => void;
+};
+
+const fetchBrowseResults = async (
+  client: ConstructorIOClient,
+  filterName: string,
+  filterValue: string,
+  browseParameters: BrowseParameters,
+  dispatch: React.Dispatch<RequestAction>,
+  // eslint-disable-next-line max-params
+) => {
+  dispatch({
+    type: RequestStatus.FETCHING,
+  });
+
+  try {
+    const res = await client.browse.getBrowseResults(filterName, filterValue, {
+      ...browseParameters,
+    });
```

Comment:
any reason we're spreading here instead of just passing the object?

Reply:
No reason probably it was copied over